Region/Region.cc

```diff
@@ -509,7 +546,7 @@ void Region::fillSpectralProfileData(CARTA::SpectralProfileData& profileData, in
             newProfile->set_coordinate(profileCoord);
             newProfile->set_stats_type(CARTA::StatsType::None);
             std::vector<float> profile;
-            if (getData(profile, sublattice)) {
+            if (getData(profile, sublattice, 100)) {
```

Comment:
Noting here that if getData returns false before the profile is complete, the SPECTRAL_PROFILE_DATA message will still be sent with empty values[ ] array.

Reply:
What we really want is a solution that checks for cancellation, _and_ sends partial data as well, so users can be reassured that progress is happening if a region takes a long time to calculate. Cancellation checks can happen at a higher cadence than the updates. 
Below is a demo of this happening on the old CyberSKA viewer (insanely slow performance due to server issues)

![region_updates](https://user-images.githubusercontent.com/592504/55350289-00789c80-54bc-11e9-9ebe-e8dfe8ddca35.gif)